src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -1517,24 +1536,133 @@ JvmtiModuleClosure::get_all_modules(JvmtiEnv* env, jint* module_count_ptr, jobje
 }
 
 void
-UpdateForPopTopFrameClosure::do_thread(Thread *target) {
+UpdateForPopTopFrameClosure::doit(Thread *target, bool self) {
+  Thread* current_thread  = Thread::current();
+  HandleMark hm(current_thread);
+  JavaThread* java_thread = target->as_Java_thread();
+
```

Comment:
This would be useful here:

`assert(_state->get_thread() == java_thread, "Must be");`